integration_tests/testsuite/test_custom.py

```diff
@@ -39,19 +57,179 @@ def runTest(self):
         if not output:
             logging.debug('No custom tests specified.')
         else:
-            for test_info in json.loads(output):
+            for specification in json.loads(output):
```

Comment:
The logic of deciding between 'simple' or 'composite' tests should be here for more readable code.